### DISCUSSION TITLE
Global Canvas Zoom with Layer Scaling

### DISCUSSION THREAD

**Discussion by iOSyan** [hm21/pro_image_editor#84] [Q&A] [answered]
Sorry, our requirements are quite complex.
Can the canvas be zoomed in or out altogether using a two-finger pinch gesture, including all the layers above?
Thank you very much for your help！

**Comment by hm21** [comment]
Currently, this function is not directly available. But you can create your own solution by using the callback that listens for scale events and then updates the layers as shown below.

```dart
mainEditorCallbacks: MainEditorCallbacks(
      onScaleUpdate: (details) {
        /// Update the layers
        editorKey.currentState!.activeLayers
     },
)
```

**Reply by iOSyan** [reply]
Sorry, I still haven't been able to implement this feature. The effect I want is something like this: 

https://github.com/hm21/pro_image_editor/assets/12389901/c012ea50-99f0-4ee1-8d6a-9a7d37933448


I want to be able to zoom the entire background canvas, and have all the layers on it zoom together. I still don't know how to achieve this effect.

**Reply by hm21** [reply]
I don't have a solution for this, but I saw in the Insights that someone made a commit where "ZoomableMainEditor" stands, so maybe [this](https://github.com/matterwavesai/pro_image_editor/tree/matterwaves) repository which he forked from me will help you.

**Reply by iOSyan** [reply]
Yes, this `ZoomableMainEditor` is the functionality I want, but it currently seems to be usable only `standalone` and cannot be used together with other features. I still need it to be combined with the example code: https://github.com/hm21/pro_image_editor/discussions/106#discussion-6829551

Will this feature be merge in the future?

**Reply by hm21** [reply, marked as answer]
Okay, I released version `4.1.0` with a minimal solution which allow zooming. You can follow this [example](https://github.com/hm21/pro_image_editor/blob/stable/example/lib/pages/zoom_move_editor_example.dart) to check how to enable/ disable it.

**Reply by iOSyan** [reply]
That's great!!!
It's so helpful!!
Thank you soooo much!
